tasks/eva_2205/src/main/groovy/eva2205/eva2205_undo_merges_into_multimaps.groovy

```diff
@@ -30,108 +37,183 @@ if (!options) {
-    dbEnv.bulkInsertIgnoreDuplicates(cvesToResurrect.findAll { it.accession >= dbsnpRSIDUpperBound }, cveClass)
+    // See https://docs.google.com/spreadsheets/d/1kRKHR4zrq-nxH_Sg82TwXZbxdgjB3K-q4YMyJva8yMg/edit#rangeid=1814908308
+    def undoMergesIntoMultiMapRS() {
+        def mapWtSSCursors = [sveClass, dbsnpSveClass].collect {
+            new RetryableBatchingCursor(where("seq").is(assembly).and("mapWeight").exists(true), dbEnv.mongoTemplate, it)
+        }
+        def collectionWithCvesToResurrect = "cvesToResurrect"
+        def svesStillWithMapWtRS = new ArrayList<>()
+        def batchIndex = 0
+        mapWtSSCursors.each {it.each {mapWtSVEs ->
+            def involvedRS = mapWtSVEs.findAll{
+                Objects.nonNull(it.clusteredVariantAccession)}.collect{it.clusteredVariantAccession}.toSet()
+            def involvedHashes = mapWtSVEs.collect{it.hashedMessage}
+            // Also pull in other SS with the same RS so that both SS and RS deprecation can take place within a batch
+            mapWtSVEs += [sveClass, dbsnpSveClass].collect{collectionClass ->
+                dbEnv.mongoTemplate.find(query(where("_id").ne(involvedHashes).and("seq").is(assembly)
+                        .and("mapWeight").exists(true).and("rs").in(involvedRS)), collectionClass)}.flatten()
+            def svesWithUndoneMultimaps = undoSVEsAssignedMultimaps(mapWtSVEs)
+            def sveHashesWithUndoneMultimaps = svesWithUndoneMultimaps.collect{
+                it.inactiveObjects[0].hashedMessage}.toSet()
+            // Restore CVEs which were incorrectly merged into map weighted CVEs
+            def cvesToResurrect = svesWithUndoneMultimaps.collect {
+                EVAObjectModelUtils.toClusteredVariantEntity(it.inactiveObjects[0].toSubmittedVariantEntity())
+            }
+            // Store these separately because these cannot be obtained again with a subsequent run in case of failures.
+            // This is because the deprecation below will take out the SVEs with mapWeight entries and that will impact mapWtSSCursor above.
+            dbEnv.bulkInsertIgnoreDuplicates(cvesToResurrect, cveClass, collectionWithCvesToResurrect)
+            // All the legitimate non map-weight RS assignments to SS have been carried out by getCVEsToResurrect above
+            // Therefore, we can proceed to make room for resurrected RS by deprecating SS associated with the current set of map-weighted RS
+            svesStillWithMapWtRS += mapWtSVEs.findAll{!sveHashesWithUndoneMultimaps.contains(it.hashedMessage)}
+            batchIndex += 1
+            // Since the deprecation job below involves the overhead of running an entire Spring batch job
+            // run it at a different frequency (at the expense of holding 100k records of svesStillWithMapWtRS in memory)
+            if (batchIndex % 10 == 0) {
+                deprecateSSBatch(svesStillWithMapWtRS)
```

Comment:
Maybe `deprecateAndClearSSBatch` or something similar, to make obvious that it modifies its input..

Reply:
Addressed..